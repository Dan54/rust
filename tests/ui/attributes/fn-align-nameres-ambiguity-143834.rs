//~ NOTE `align` could refer to a built-in attribute

// Anti-regression test to demonstrate that at least we mitigated breakage from adding a new
// `#[align]` built-in attribute.

// Needs edition >= 2018 macro use behavior.
//@ edition: 2018

macro_rules! align {
    //~^ NOTE `align` could also refer to the macro defined here
    () => {
        /* .. */
    };
}

pub(crate) use align;
//~^ ERROR `align` is ambiguous
//~| NOTE ambiguous name
//~| NOTE ambiguous because of a name conflict with a builtin attribute

fn main() {}
