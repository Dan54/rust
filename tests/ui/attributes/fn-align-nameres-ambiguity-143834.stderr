error[E0659]: `align` is ambiguous
  --> $DIR/fn-align-nameres-ambiguity-143834.rs:16:16
   |
LL | pub(crate) use align;
   |                ^^^^^ ambiguous name
   |
   = note: ambiguous because of a name conflict with a builtin attribute
   = note: `align` could refer to a built-in attribute
note: `align` could also refer to the macro defined here
  --> $DIR/fn-align-nameres-ambiguity-143834.rs:9:1
   |
LL | / macro_rules! align {
LL | |
LL | |     () => {
LL | |         /* .. */
LL | |     };
LL | | }
   | |_^

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0659`.
